design-proposals/garbage-collect-completed-dvs.md

```diff
@@ -0,0 +1,70 @@
+# Overview
+DataVolumes are a useful construct for provisioning VM disks but once the PVC is provisioned and populated the DataVolume provides no value. This design presents an approach for garbage collecting completed DataVolumes. 
+
+## Motivation
+In the field, customers are confused by DVs after the fact. For example, when someone deletes a PVC that is owned by a DV they are confused when the PVC is re-created.
+
+## Goals
+DataVolumes will be garbage collected after their PVC is provisioned and populated.
+
+## Non Goals
+(limitations to the scope of the design)
+
+## Definition of Users
+(who is this feature set intended for)
+
+## User Stories
+* As a VM owner I use a standalone DataVolume to prepare a VM disk PVC for use by my VM. Once the DataVolume is complete, it has no use and can be confusing, so I want the system to garbage collect it. The PVC should still maintain an independent lifecycle from the VM.
+* As a VM owner I add an entry to the dataVolumeTemplates section of my VM spec to have a disk prepared for my VM. Once the DataVolume is complete, it has no use and can be confusing, so I want the system to garbage collect it. I want kubevirt to understand that it doesn't need to create a new DataVolume when the underlying PVC exists. The PVC should still share the same lifecycle as the VM.
+* As a VM owner I want to restore my backed up VM without needing to recreate the DataVolume.
+* As a VM owner I want to replicate my workload to another cluster without needing to mutate the PVC and DV with special annotations in order for them to behave as expected in the new cluster.
+
+## Repos
+* **CDI**: Controllers (DataVolume/Import/Upload/Clone/DataImportCron/DataSource) will need adaptations:
+** So far the controllers assumed the existance of DV-PVC pairs, which is no longer a valid assumption.
+** DataImportCron controller currently re-creates the import DV if deleted, even if the PVC was already populated.
+** Warm imports - verify DV is Succeded only after the last cycle
+* **KubeVirt**: VM controller adaptations (see below)
+* **Tekton tasks** - check how DVs are used to make sure GC by default wouldn't break them
+* **UI**: check if DVs are referred after Succeded
+* **HCO**: expose CDIConfig `dataVolumeTTL`
+
+## Configuration
+* CDIConfig - minimal time for DV garbage collection after completion (disabled by default, for opt-in)
+  `dataVolumeTTL *int32`
```

Comment:
Is this seconds/minutes/hours/milliseconds?

Reply:
Seconds. Better rename to `dataVolumeTTLSeconds`?

Reply:
Yeah that way people know the quantity to enter.